src/layers/core/hexagon-layer/hexagon-layer.js

```diff
@@ -0,0 +1,221 @@
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+import {Layer} from '../../../lib';
+import {assembleShaders} from '../../../shader-utils';
+import {Model, CylinderGeometry} from 'luma.gl';
+import {readFileSync} from 'fs';
+import {join} from 'path';
+import assert from 'assert';
+import {mat4} from 'gl-matrix';
+
+function positionsAreEqual(v1, v2) {
+  // Hex positions are expected to change entirely, not to maintain some
+  // positions and change others. Right now we only check a single vertex,
+  // because H3 guarantees order, but even if that wasn't true, this would only
+  // return a false positive for adjacent hexagons, which is close enough for
+  // our purposes.
+  return v1 === v2 || (
+    v1 && v2 && v1[0][0] === v2[0][0] && v1[0][1] === v2[0][1]
+  );
+}
+
+const defaultProps = {
+  id: 'hexagon-layer',
+  dotRadius: 1,
+  elevation: 0,
+  enable3d: true,
+  invisibleColor: {r: 0, g: 0, b: 0},
+  getCentroid: x => x.centroid,
+  getColor: x => x.color,
+  getAlpha: x => 255,
+  getElevation: x => x.elevation
+};
+
+// viewMatrix added as Uniform for lighting calculation
+const viewMatrixCompat = mat4.create();
+mat4.lookAt(viewMatrixCompat, [0, 0, 0], [0, 0, -1], [0, 1, 0]);
+const viewMatrix = new Float32Array(viewMatrixCompat);
+
+export default class HexagonLayer extends Layer {
+
+  /**
+   * @classdesc
+   * HexagonLayer is a variation of grid layer, it is intended to render
+   * hexagon tessellations. It supports elevation, lighting as well
+   *
+   * @class
+   * @param {object} props
+   * @param {number} props.dotRadius - hexagon radius multiplier
+   * @param {number} props.elevation - hexagon elevation uniform, 1 unit roughly equals 100 meters
+   * @param {boolean} props.enable3d - if set to false, all hexagons will be flat
+   * @param {array} props.hexagonVertices - primitive hexagon vertices as [[lon, lat]]
+   * @param {object} props.invisibleColor - hexagon invisible color
+   * @param {function} props.getCentroid - hexagon centroid should be formatted as [lon, lat]
+   * @param {function} props.getColor -  hexagon color should be formatted as [255, 255, 255]
+   * @param {function} props.alpha -  hexagon opacity should be from 0 - 255
+   * @param {function} props.getElevation - hexagon elevation 1 unit approximate to 100 meters
+   *
+   */
+  constructor(props) {
+    assert(props.hexagonVertices, 'hexagonVertices must be supplied');
+    super(Object.assign({}, defaultProps, props));
+  }
+
+  /**
+   * DeckGL calls initializeState when GL context is available
+   * Essentially a deferred constructor
+   */
+  initializeState() {
+    const {gl} = this.context;
+    this.setState({model: this.getModel(gl)});
+
+    const {attributeManager} = this.state;
+    attributeManager.addInstanced({
+      instancePositions: {size: 3, update: this.calculateInstancePositions},
```

Comment:
Note for upcoming API audit:
In this case it is possible that splitting into `positions` and `height` (getPositions and getHeight) would make more sense.
 E.g. being able to animate height as well as color for a set of hexagons.